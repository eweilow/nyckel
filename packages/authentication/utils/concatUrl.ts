import { URL } from "url";

export function concatUrl(origin: string, path: string) {
  const url = new URL(origin);
  url.pathname = path;

  if (!url.protocol) url.protocol = "http";

  return new URL(url.href).href;
}
