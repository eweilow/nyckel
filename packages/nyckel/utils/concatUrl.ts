import { URL } from "url";

export function concatUrl(origin: string, path: string) {
  const url = new URL(origin);
  url.pathname = path;

  return url.href;
}
