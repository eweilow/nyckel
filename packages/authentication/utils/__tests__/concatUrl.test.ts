import { concatUrl } from "../concatUrl";

describe("concatUrl", () => {
  it("should work correctly", () => {
    const cases: [string, string][] = [
      ["https://example.com", "/test/test"],
      ["https://example.com", "https://example.com"],
      ["https://example.com", ""],
      ["https://example.com:3000", ""],
      ["http://example.com:3000", ""],
      ["http://example.com", ""]
    ];

    for (const [origin, pathname] of cases) {
      const url = concatUrl(origin, pathname);

      const parsed = new URL(url);
      expect({
        input: {
          origin,
          pathname
        },
        output: {
          url,
          origin: parsed.origin,
          pathname: parsed.pathname
        }
      }).toMatchSnapshot();
    }
  });
});
