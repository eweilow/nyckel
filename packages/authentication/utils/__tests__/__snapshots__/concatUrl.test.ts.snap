// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`concatUrl should work correctly 1`] = `
Object {
  "input": Object {
    "origin": "https://example.com",
    "pathname": "/test/test",
  },
  "output": Object {
    "origin": "https://example.com",
    "pathname": "/test/test",
    "url": "https://example.com/test/test",
  },
}
`;

exports[`concatUrl should work correctly 2`] = `
Object {
  "input": Object {
    "origin": "https://example.com",
    "pathname": "https://example.com",
  },
  "output": Object {
    "origin": "https://example.com",
    "pathname": "/https://example.com",
    "url": "https://example.com/https://example.com",
  },
}
`;

exports[`concatUrl should work correctly 3`] = `
Object {
  "input": Object {
    "origin": "https://example.com",
    "pathname": "",
  },
  "output": Object {
    "origin": "https://example.com",
    "pathname": "/",
    "url": "https://example.com/",
  },
}
`;

exports[`concatUrl should work correctly 4`] = `
Object {
  "input": Object {
    "origin": "https://example.com:3000",
    "pathname": "",
  },
  "output": Object {
    "origin": "https://example.com:3000",
    "pathname": "/",
    "url": "https://example.com:3000/",
  },
}
`;

exports[`concatUrl should work correctly 5`] = `
Object {
  "input": Object {
    "origin": "http://example.com:3000",
    "pathname": "",
  },
  "output": Object {
    "origin": "http://example.com:3000",
    "pathname": "/",
    "url": "http://example.com:3000/",
  },
}
`;

exports[`concatUrl should work correctly 6`] = `
Object {
  "input": Object {
    "origin": "http://example.com",
    "pathname": "",
  },
  "output": Object {
    "origin": "http://example.com",
    "pathname": "/",
    "url": "http://example.com/",
  },
}
`;
